compiler/resolution/functionResolution.cpp

```diff
@@ -5637,6 +5640,10 @@ resolveExpr(Expr* expr) {
       }
     }
 
+    if (!tryFailure)
+      callStack.pop();
```

Comment:
I'm having trouble understanding why callStack.pop() needs to move. Is this change necessary for ForallStmts? Is this just trying to make this code clearer? If so it should not be in this PR.